### PR TITLE
Persistence fixes

### PR DESCRIPTION
**Pull request by zbynek001**
some small persistence fixes:

* Enforce valid seqnr for deletes, migrated from https://github.com/akka/akka/pull/25488

* lastSequenceNr should reflect the snapshot sequence and not start with 0 when journal is empty. Migrated from https://github.com/akka/akka/pull/27496

* snapshot RecoveryTick ignored, part of https://github.com/akka/akka/pull/20753

### CODE REVIEW

File: src/core/Akka.Persistence.Tests/PersistentActorSpec.Actors.cs [hunk 1]
```diff
@@ -39,6 +39,13 @@ protected bool Receiver(object message)
                     throw new ArgumentNullException("Received DeleteMessagesSuccess without anyone asking for delete!");
                 AskedForDelete.Tell(message);
             }
+            else if (message is DeleteMessagesFailure)
+            {
+                if (AskedForDelete == null)
+                    throw new ArgumentNullException("Received DeleteMessagesSuccess without anyone asking for delete!");
```
Comment (Aaronontheweb) [inline src/core/Akka.Persistence.Tests/PersistentActorSpec.Actors.cs:45]: Error message is wrong here

File: src/core/Akka.Persistence.Tests/PersistentActorSpec.cs [hunk 2]
```diff
@@ -604,6 +605,20 @@ public void PersistentActor_should_be_able_to_delete_all_events()
             ExpectMsg<object[]>(m => m.Length == 0);
         }
 
+        [Fact]
+        public void PersistentActor_should_not_be_able_to_delete_higher_seqnr_than_current()
```
Comment (Aaronontheweb) [inline src/core/Akka.Persistence.Tests/PersistentActorSpec.cs:609]: LGTM

File: src/core/Akka.Persistence/Eventsourced.Recovery.cs [hunk 3]
```diff
@@ -208,8 +208,9 @@ private EventsourcedState Recovering(Receive recoveryBehavior, TimeSpan timeout)
                         case RecoverySuccess success:
                             timeoutCancelable.Cancel();
                             OnReplaySuccess();
-                            _sequenceNr = success.HighestSequenceNr;
-                            LastSequenceNr = success.HighestSequenceNr;
+                            var highestSeqNr = Math.Max(success.HighestSequenceNr, LastSequenceNr);
```
Comment (Aaronontheweb) [inline src/core/Akka.Persistence/Eventsourced.Recovery.cs:211]: LGTM

**Review by Aaronontheweb — APPROVED** [review]
LGTM

File: src/core/Akka.Persistence/Eventsourced.Recovery.cs (hunk 3 above)
Comment (ismaelhamed) [reply src/core/Akka.Persistence/Eventsourced.Recovery.cs:211]: These changes are making the tests for `Akka.Persistence.Oracle` to fail. Reverting them fixes it. 

The original PR also had a `SnapshotRecoveryWithEmptyJournalSpec` test that we're missing and could shed some light.

Comment (ismaelhamed) [reply src/core/Akka.Persistence/Eventsourced.Recovery.cs:211]: Nevermind, it was a problem with the Oracle plugin itself.

**Pull request merged**